scripts/src/utils/RateLimit.ts

```diff
@@ -0,0 +1,101 @@
+import * as moment from "moment";
+
+import { getRedisClient, RedisAsyncClient } from "../redis";
+import { TooManyRegistrations, TooMuchEarnOrdered } from "../errors";
+
+class RateLimit {
+	private readonly bucketPrefix: string;
+	private readonly rateLimitValue: number;
+	private readonly redis: RedisAsyncClient;
+	private windowSize: number = 0;
+	private bucketSize: number = 0;
+	private currentTimestampSeconds: number = 0;
+	private ttl: number = 172800; // two days in seconds
```

Comment:
maybe use `moment.duration(days=2).asSeconds()`?

Reply:
Don't forget the ttl will be evaluated on every constructor run :)